### PR TITLE
Handle zero division and remainder

### PR DESCRIPTION
**Pull request by werat**

### CODE REVIEW

File: lldb-eval/eval.cc [hunk 1]
```diff
@@ -686,6 +687,52 @@ Value Interpreter::EvaluateSubtraction(Value& lhs, Value& rhs) {
...
+                      rhs_scalar.type_ == Scalar::Type::DOUBLE;
+
+  if (!is_float_div && rhs_scalar.GetInt64() == 0) {
+    // This is UB and the compiler would generate a warning:
+    //
+    //  warning: division by zero is undefined [-Wdivision-by-zero]
+    //
+    return Value(Scalar(0));
+  }
+
```
Comment (octurion) [inline lldb-eval/eval.cc:707]: You also need to handle `INT_MIN / -1` (which results in a SIGFPE).

File: lldb-eval/eval.cc [hunk 2]
```diff
@@ -686,6 +687,52 @@ Value Interpreter::EvaluateSubtraction(Value& lhs, Value& rhs) {
...
+  if (rhs_scalar.GetInt64() == 0) {
+    // This is UB and the compiler would generate a warning:
+    //
+    //  warning: remainder by zero is undefined [-Wdivision-by-zero]
+    //
+    return Value(Scalar(0));
+  }
+
+  return Value(lhs_scalar % rhs_scalar);
```
Comment (octurion) [inline lldb-eval/eval.cc:733]: Ditto for `INT_MIN % -1`.

File: lldb-eval/eval.cc [hunk 3]
```diff
@@ -686,6 +687,52 @@ Value Interpreter::EvaluateSubtraction(Value& lhs, Value& rhs) {
...
+  }
+
+  auto lhs_scalar = lhs.AsScalar();
+  auto rhs_scalar = rhs.AsScalar();
+
+  if (rhs_scalar.GetInt64() == 0) {
+    // This is UB and the compiler would generate a warning:
+    //
+    //  warning: remainder by zero is undefined [-Wdivision-by-zero]
+    //
```
Comment (octurion) [inline lldb-eval/eval.cc:729]:
```suggestion
```

**Comment by werat** [comment]
Closed in favor of #28

**Pull request closed without merging**